akka-http-core/src/test/scala/akka/http/impl/util/ExampleHttpContexts.scala

```diff
@@ -62,4 +69,14 @@ object ExampleHttpContexts {
 
   def loadX509Certificate(resourceName: String): Certificate =
     CertificateFactory.getInstance("X.509").generateCertificate(resourceStream(resourceName))
+
+  /**
+   * A client transport that will rewrite the target address to a fixed address. This can be used
+   * to pretend to connect to akka.example.org which is required to connect to the example server certificate.
+   */
+  def proxyTransport(realAddress: InetSocketAddress): ClientTransport =
```

Comment:
Ported the trick we used in alpakka to be to test TLS hostname verification in tests.